src/org/toml/lang/icons/TomlIconProvider.kt

```diff
@@ -0,0 +1,25 @@
+package org.toml.lang.icons
+
+import com.intellij.ide.IconProvider
+import com.intellij.psi.PsiElement
+import org.toml.lang.core.psi.TomlFile
+import javax.swing.Icon
+
+
+public class TomlIconProvider : IconProvider() {
+    override fun getIcon(element: PsiElement, flags: Int): Icon? {
+        return when(element) {
+            is TomlFile -> getFileIcon(element)
+            else -> null
+        }
+    }
+
+    fun getFileIcon(element: TomlFile): Icon? {
```

Comment:
There is currently no one calling the method externally, but I don't see why that shouldn't happen if you already have a `TomlFile` instance and don't want to perform the unnecessary pattern matching in the `getIcon()` method